.github/workflows/windows.yml

```diff
@@ -12,70 +12,56 @@ jobs:
     name: Build and test (Windows, mingw-w64)
     runs-on: windows-latest
     steps:
-      - name: Checkout Git branch
-        uses: actions/checkout@v2
-
-      - name: Cache Msys2
-        id: cache-msys2
-        uses: actions/cache@v1.1.2
-        with:
-          path: msys64
-          key: ${{ runner.os }}-msys64
-
-      - name: Get Msys2
-        if: steps.cache-msys2.outputs.cache-hit != 'true'
-        run: |
-          Invoke-WebRequest -Outfile msys2.tar.xz -Uri "https://github.com/msys2/msys2-installer/releases/download/nightly-x86_64/msys2-base-x86_64-latest.tar.xz"
-          xz -d msys2.tar.xz
-          tar -xf msys2.tar
-          rm msys2.tar
 
       - name: Add Msys2 to Path
         run: |
-          echo "::add-path::${env:GITHUB_WORKSPACE}/msys64/mingw64/bin"
-          echo "::add-path::${env:GITHUB_WORKSPACE}/msys64/usr/bin"
+          echo "::add-path::C:/msys64/usr/bin"
+          echo "::add-path::C:/msys64/mingw64/bin"
 
       - name: Initialize msys2
         shell: cmd
-        if: steps.cache-msys2.outputs.cache-hit != 'true'
         run: bash -lc 'exit'
 
       - name: Update Pacman
         run: pacman -Syy
 
       - name: Cache pacman Packages
         id: cache-pacman
-        uses: actions/cache@v1.1.2
+        uses: actions/cache@v2
         with:
-          path: msys64/var/cache/pacman/pkg
+          path: C:/msys64/var/cache/pacman/pkg
           key: ${{ runner.os }}-pacman
 
       - name: Update Msys2 Packages
-        run: pacman -Suu --needed --ask=20 --noconfirm
+        run: |
+          pacman -R --noconfirm mingw-w64-x86_64-gcc-ada mingw-w64-x86_64-gcc-objc mingw-w64-x86_64-gcc-fortran mingw-w64-x86_64-gcc-libgfortran
```

Comment:
Sure, i guess this would have definitely warranted a comment

The reason why these are being removed is because upstream dropped these packages when updating gcc because of a bug fix, so trying to upgrade gcc with the current package version results in errors of unresolved dependencies

Reply:
That sounds unfortunate. Can you add a comment like that to the file? That would be the last thing before merge in my eyes. Thank you!

Reply:
Actually, just a second, going to reconfirm if that is the case, since it could be I just need to add a `pacman -Suu` right before to make sure to upgrade all of the other packages and it could just be that the discussion that I read before was old news (nearly a month ago)

Reply:
Would be great to re-verify, thanks!

Reply:
Hmm, although I could upgrade and then install, it seems it will be much faster to just go ahead and remove the other packages and then just upgrade gcc etc (6+48 seconds vs 12 minutes, github actions just added msys2 support and it's already a bit outdated.)

Reply:
Going to just mention that in the comment instead

Reply:
done